Source/WTF/wtf/SuspendableWorkQueue.cpp

```diff
@@ -120,18 +138,26 @@ void SuspendableWorkQueue::invokeAllSuspensionCompletionHandlers()
 void SuspendableWorkQueue::suspendIfNeeded()
 {
     ASSERT(!isMainThread());
-
     Locker suspensionLocker { m_suspensionLock };
+
     auto suspendFunction = std::exchange(m_suspendFunction, { });
-    if (m_state != State::WillSuspend)
+    if (m_state != State::WillSuspend) {
+        if (m_state == State::Suspended) {
+            // If state is suspended, the queue should not be running.
+            RELEASE_LOG_ERROR_IF(m_shouldLog, SuspendableWorkQueue, "%p - SuspendableWorkQueue::suspendIfNeeded current state Suspended", this);
```

Comment:
Could use `RELEASE_LOG_ERROR_IF(m_shouldLog && m_state == State::Suspended, ` to make the code a bit more concise.